pubsub.tf

```diff
@@ -2,7 +2,14 @@ resource "google_pubsub_topic" "topic" {
   project = var.project
   name    = var.topic
 
-  labels = var.labels
+  labels = local.labels
+
+  lifecycle {
+    precondition {
+      condition     = var.shared == false ? var.team != null : var.team == null
+      error_message = "If var.shared = false, then var.team must be set to valid team name"
+    }
```

Comment:
yes, I know this will only trigger on plan (potentially wasting a few seconds), but you cannot reference another variable inside another variable's validation block :))